std::any breaks gr_modtool bind

### What happened?

The commit https://github.com/gnuradio/gnuradio/commit/041f44e809f39d0567f6f191284d555cd6d08de2 breaks gr_modtool bind.

### System Information

OS: Ubuntu 20.04
GR Installation Method: Source


### GNU Radio Version

3.10-git (master)

### Specific Version

tip of master

### Steps to Reproduce the Problem

In an OOT:

gr_modtool bind module_name

### Relevant log output

```shell
e@w6rz:~/xfer/gr-atsc3$ gr_modtool bind modulator_bc
GNU Radio module name identified: atsc3
INFO Parsing source file "/home/re/xfer/gr-atsc3/include/atsc3/modulator_bc.h" ... 
In file included from /home/re/xfer/gr-atsc3/include/atsc3/modulator_bc.h:13:
In file included from /opt/gnuradio-3.10.0git/include/gnuradio/block.h:17:
In file included from /opt/gnuradio-3.10.0git/include/gnuradio/basic_block.h:15:
In file included from /opt/gnuradio-3.10.0git/include/gnuradio/io_signature.h:17:
In file included from /opt/gnuradio-3.10.0git/include/gnuradio/buffer_double_mapped.h:15:
In file included from /opt/gnuradio-3.10.0git/include/gnuradio/buffer.h:18:
In file included from /opt/gnuradio-3.10.0git/include/gnuradio/tags.h:15:
/opt/gnuradio-3.10.0git/include/pmt/pmt.h:677:35: error: no type named 'any' in
      namespace 'std'
PMT_API pmt_t make_any(const std::any& any);
                             ~~~~~^
/opt/gnuradio-3.10.0git/include/pmt/pmt.h:680:14: error: no type named 'any' in
      namespace 'std'
PMT_API std::any any_ref(pmt_t obj);
        ~~~~~^
/opt/gnuradio-3.10.0git/include/pmt/pmt.h:683:44: error: no type named 'any' in
      namespace 'std'
PMT_API void any_set(pmt_t obj, const std::any& any);
                                      ~~~~~^
3 errors generated.
Error occurred while running CASTXML:  status:1
re@w6rz:~/xfer/gr-atsc3$
```
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,6 +54,7 @@ set(GR_CMAKE_MIN_VERSION "3.10.2")    ## Version in Ubuntu 18.04LTS
 set(GR_MAKO_MIN_VERSION "1.0.7")      ## debian buster, 18.04LTS
 set(GR_PYTHON_MIN_VERSION "3.6.5")    ## Version in Ubuntu 18.04LTS
 set(GR_NUMPY_MIN_VERSION "1.13.3")    ## Version in Ubuntu 18.04LTS
+set(GR_PYGCCXML_MIN_VERSION "2.0.0")  ## Version to support c++17 (in pip)
 set(GCC_MIN_VERSION "8.3.0")          ## debian buster
 set(CLANG_MIN_VERSION "11.0.0")       ## debian bullseye, Fedora 33
 set(APPLECLANG_MIN_VERSION "1100")    ## same as clang 11.0.0, in Xcode11
@@ -324,12 +325,20 @@ GR_PYTHON_CHECK_MODULE(
     "LooseVersion(numpy.__version__) >= LooseVersion('${GR_NUMPY_MIN_VERSION}')"
     NUMPY_FOUND)
 # Needed for automatic regeneration of some bindings
-GR_PYTHON_CHECK_MODULE_RAW(
+GR_PYTHON_CHECK_MODULE(
     "pygccxml"
-    "import pygccxml"
+    pygccxml
+    "LooseVersion(pygccxml.__version__) >= LooseVersion('${GR_PYGCCXML_MIN_VERSION}')"
     PYGCCXML_FOUND
     )
-
+if(NOT PYGCCXML_FOUND)
+    message(STATUS "")
+    message(STATUS "***************************** WARNING!!! *************************")
+    message(STATUS "pygccxml is highly recommended for using gr_modtool")
+    message(STATUS "and is either not present or below the minimum version " ${GR_PYGCCXML_MIN_VERSION})
+    message(STATUS "Only trivial bindings will be generated using gr_modtool bind ")
+    message(STATUS "******************************************************************")
+endif()
 
 find_package(pybind11 REQUIRED)
 IF(${pybind11_VERSION} VERSION_LESS ${PYBIND11_MIN_VERSION})

--- a/gr-utils/blocktool/core/parseheader_generic.py
+++ b/gr-utils/blocktool/core/parseheader_generic.py
@@ -326,7 +326,7 @@ class GenericHeaderParser(BlockTool):
                 compiler='gcc',
                 undefine_symbols=['__PIE__'],
                 define_symbols=self.define_symbols,
-                cflags='-std=c++11 -fPIC')
+                cflags='-std=c++17 -fPIC')
             decls = parser.parse(
                 [self.target_file], xml_generator_config)
 

